portal/portal-service-impl/impl/src/java/org/sakaiproject/portal/service/BullhornServiceImpl.java

```diff
@@ -613,4 +689,53 @@ private boolean clearAllAlerts(String alertType, String userId) {
         countCache.remove(userId);
         return true;
     }
+
+    public boolean toggleHideBullhornAlerts(final String userId) {
+        try {
+            PreferencesEdit prefEdit = this.preferencesService.edit(userId);
+            ResourcePropertiesEdit props = prefEdit.getPropertiesEdit();
+            boolean hideAlerts;
+            try {
+                hideAlerts = !props.getBooleanProperty(ALERTS_HIDE_PROP);
+            } catch (EntityPropertyNotDefinedException | EntityPropertyTypeException e) {
+                // Hide Alerts preference was never set or is not a boolean value
+                // So the user must have clicked the "Hide" button
+                hideAlerts = true;
+            }
+            props.addProperty(ALERTS_HIDE_PROP, String.valueOf(hideAlerts));
+            this.preferencesService.commit(prefEdit);
+            return hideAlerts;
+        } catch (PermissionException e) {
+            log.error("Unable to toggle bullhorn alert preferences.", e);
+        } catch (InUseException e) {
+            log.error("Unable to edit user preferences.", e);
+        } catch (IdUnusedException e) {
+            log.error("Preferences have never been set for user: ", userId, e);
+        }
+        // No permissions or unable to edit user preferences. This shouldn't occur
+        // let the user see notification alerts.
+        return false;
+    }
+
+    public boolean isAlertHidden(final String userId) {
+        try {
+            Preferences prefs = this.preferencesService.getPreferences(userId);
+            ResourceProperties props = prefs.getProperties();
+            return props.getBooleanProperty(ALERTS_HIDE_PROP);
+        } catch (EntityPropertyTypeException | EntityPropertyNotDefinedException e) {
+            return false;
+        }
+    }
+
+    private void switchToAdmin() {
+
+        org.sakaiproject.tool.api.Session session = sessionManager.getCurrentSession();
+        session.setUserId(UserDirectoryService.ADMIN_ID);
+    }
+
+    private void switchToNull() {
+
+        org.sakaiproject.tool.api.Session session = sessionManager.getCurrentSession();
+        session.setUserId(null);
```

Comment:
This method will become unnecessary if you use a `SecurityAdvisor` pattern rather than switching the session's user ID to admin.